azavea/raster-foundry
# Add Rollbar integration to front-end

## Overview

This PR adds Rollbar to the front-end.

As it is set right now, it will forward all uncaught errors to Rollbar. We can also manually trigger errors of various levels.  See https://rollbar.com/docs/notifier/rollbar.js/ for details on this.

The front-end gets the necessary credentials from the `config` endpoint. Only errors in staging and production are forwarded to Rollbar.

EDIT:

This PR now has a `rollbarWrapperService` that allows for setting the Rollbar configuration from various parts in the app in a consistent way.

`rollbarWrapperService.init(user:Object)` can take an optional user profile as an argument and set the Rollbar configuration appropriately. Calling the method without an argument removes the user info from the configuration.

The user data is passed through the `person` key of the Rollbar payload. In Rollbar, under the 'occurences' tab of an error, this data is visible if it was sent with the error.

EDIT: Also wrapped in a quick fix of the trailing slash default route that has been causing some problems.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

If you look in the Raster Foundry Slack or Rollbar,  you'll see that I had sent an error there via the front-end "TEST:Something bad happened".

### Notes

To make this work, you will need to update your `.env` file. `./scripts/bootstrap` will do this, or you can extract the aws-cli command from the bootstrap script and get the new `.env` without having to go through the entire bootstrap process.


## Testing Instructions

 * Make sure your `.env` has `ROLLBAR_CLIENT_TOKEN` defined
 * Start up the server and front-end, making sure your dev tools are recording requests.
 * After the front-end boots, find the request that fetched the `config` endpoint and see that the `rollbarClientToken` is included in the response.
 * Trigger an error, either by inducing one with bad JS or by manually triggering one from the console (the `Rollbar` object is available globally and you can use the `Rollbar.error` function). Check that this error was not logged in Rollbar (b/c it is under the development env).
 * In `index.config.js`, change line 73 to hardcode the environment within the `RollbarProvider.init` function to 'staging'
* Trigger an error again, and see that it is logged to Rollbar

Closes #1306
